app-creator/client/src/data/templates.ts

```diff
@@ -0,0 +1,54 @@
+import { TemplateItem, database } from '../client/fetch-templates';
+
+class TemplatesManager {
+  private templates: TemplateItem[] = [];
+
+  getTemplates() {
+    return this.templates;
+  }
+
+  async fetchTemplates(
```

Comment:
I would recommend adding a JSDoc comment at the top of the method to explain the fallback behavior.